### PR TITLE
 Migrate to new lit*

### LINKED ISSUE #199: Migrate to new lit* once it's stable
* Maybe we could move our mixing to controllers
* Maybe we could use some refs
* Try the react wrapper stuff
* Use Lit labs motion (animation)

https://lit.dev/docs/releases/upgrade/

TODO:

* [x] Replace all `lit-element` and `lit-html` with `lit`
* [x] (cancelled because of lit-analyzer) `static get properties () {` => `static properties = {`
* [x] (cancelled because of lit-analyzer) `static get styles () {` => `static styles = {`
* [x] Use `willUpdate` instead of `update`
  * [x] Make sure we don't call super with `willUpdate`
* [x] Make sure we don't call super with `updated`
* [x] _private => `state: true` (and remove the attribute false)
* [x] Remove the shim in the CDN build
* [x] Check the minification system
* [x] Check if HMR still works
* [x] Check `eslint-plugin-lit`
* [x] Check lit-analyzer
* [x] Check CEM analyzer
* [x] Check npm build
* [x] Check CDN build
* [x] Test all components

### PR DESCRIPTION
**Pull request by hsablonniere**
Fixes #199

### CODE REVIEW

**Review by pdesoyres-cc — CHANGES_REQUESTED** [review]
AST is life! :exploding_head:

Eslint check doesn't pass. It complains on:
```
static properties = {
	...
};
```

**Comment by hsablonniere** [comment]
> AST is life! exploding_head
> 
> Eslint check doesn't pass. It complains on:
> 
> ```
> static properties = {
> 	...
> };
> ```

I rolled back this change because of lit-analyzer

File: package.json [hunk 1]
```diff
@@ -42,9 +42,10 @@
     "@shoelace-style/shoelace": "^2.0.0-beta.47",
     "chart.js": "^3.5.0",
     "chartjs-plugin-datalabels": "^2.0.0",
+    "estree-walker": "^3.0.1",
     "leaflet": "^1.7.1",
-    "lit-element": "^2.5.1",
-    "lit-html": "^1.4.1",
+    "lit": "^2.2.8",
```
[the remark this answers is not in the document]
Comment (hsablonniere) [reply package.json]: My branch was not up to date. We're now on 2.3.1 ;-)

File: docs/cc-example-component.js [hunk 2]
```diff
@@ -191,42 +191,38 @@ export class CcExampleComponent extends LitElement {
...
-        /*region SMALL*/
-        .foobar.small {
-          color: green;
-        }
-
-        /*endregion*/
-      `,
-    ];
-  }
+  static styles = [
```
[the remark this answers is not in the document]
Comment (hsablonniere) [reply docs/cc-example-component.js]: I rolled back this change because of lit-analyzer so the comment is back ;-)

**Pull request merged**